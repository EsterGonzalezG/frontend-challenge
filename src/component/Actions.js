import React from 'react';

export const Actions = ({ product, setColor, setStorage }) => (
  <div>
    {
      <form action=''>
        <div className='color_container'>
          <label htmlFor='color'>Color:</label>
          <select name='color' id='color' value={product.colors} onChange={(event) => setColor(event.target.value)}>
            {product.options
              ? product.options.colors.map((color, index) =>
                  index === 0 ? (
                    <option value={color.code} key={index} defaultValue={color.name}>
                      {color.name}
                    </option>
                  ) : (
                    <option value={color.code} key={index}>
                      {color.name}
                    </option>
                  ),
                )
              : ''}
          </select>
        </div>
        <div className='storage_container'>
          <label htmlFor='storage'>Almacenamiento:</label>
          <select
            name='storage'
            id='storage'
            value={product.storage}
            onChange={(event) => setStorage(event.target.value)}
          >
            {product.options
              ? product.options.storages.map((storage, index) =>
                  index === 0 ? (
                    <option value={storage.code} key={index} defaultValue={storage.name}>
                      {storage.name}
                    </option>
                  ) : (
                    <option value={storage.code} key={index}>
                      {storage.name}
                    </option>
                  ),
                )
              : ''}
          </select>
        </div>
      </form>
    }
    <button>AÃ±adir</button>
  </div>
);
