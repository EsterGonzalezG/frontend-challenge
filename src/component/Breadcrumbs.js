import React from 'react';
import { Link, useLocation } from 'react-router-dom';

export const Breadcrumbs = React.memo(() => {
  const location = useLocation();
  const config = {
    home: [{ label: 'Home', path: '/' }],
    details: [{ label: 'Home', path: '/' }, { label: 'Product details' }],
  };

  function crumb() {
    if (location.pathname === '/') {
      return renderCrumb(config.home);
    } else {
      return renderCrumb(config.details);
    }
  }

  function renderCrumb(option) {
    return option.map(({ label, path }, index) => {
      return option.length - 1 === index ? (
        <li className='breadcrumbs-item' key={index}>
          {label}
        </li>
      ) : (
        <li key={index}>
          <Link to={path} className='breadcrumbs-link'>
            {label}
          </Link>
        </li>
      );
    });
  }

  return (
    <nav>
      <ul className='breadcrumbs color-dark-60'>{crumb()}</ul>
    </nav>
  );
});
