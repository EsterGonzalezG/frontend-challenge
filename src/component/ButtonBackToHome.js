import { PropTypes } from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';

export const ButtonBackToHome = ({ children }) => (
  <Link className='' to='/'>
    {children}
  </Link>
);
ButtonBackToHome.prototype = {
  children: PropTypes.string.isRequired,
};
