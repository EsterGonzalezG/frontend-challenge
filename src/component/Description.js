import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import { setTitleDescription, Subtitle } from './index';

export const Description = React.memo(({ description }) => {
  const [descriptionValue, setDescriptionValue] = useState([]);
  useEffect(() => {
    setDescriptionValue(setTitleDescription(description));
  }, [description]);

  return (
    <>
      <Subtitle>Characteristics:</Subtitle>
      <ul className='color-dark-100 l-marginBottom-24 font-m'>
        {descriptionValue.map(({ id, value }) => (
          <li key={id} className='color-violet'>
            {id} :
            {value !== '' ? <span className='color-dark-60'>{value}</span> : <span className='color-dark-60'> - </span>}
          </li>
        ))}
      </ul>
    </>
  );
});

Description.prototype = {
  description: PropTypes.shape({
    brand: PropTypes.string,
    model: PropTypes.string,
    price: PropTypes.string,
    cpu: PropTypes.string,
    ram: PropTypes.string,
    os: PropTypes.string,
    displayResolution: PropTypes.string,
    battery: PropTypes.string,
    primaryCamera: PropTypes.string,
    secondaryCmera: PropTypes.string,
    dimentions: PropTypes.string,
    weight: PropTypes.string,
  }).isRequired,
};
