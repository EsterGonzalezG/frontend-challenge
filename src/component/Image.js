import React from "react";
import PropTypes from 'prop-types';


export const Image = ({ item }) => (
    <img loading="lazy" className='l-marginBottom-16' src={item.imgUrl} alt={item.title}/>
)

Image.prototype = {
    imgUrl: PropTypes.string,
    title: PropTypes.string
}