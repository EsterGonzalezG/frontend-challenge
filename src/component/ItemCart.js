import { PropTypes } from 'prop-types';
import React from 'react';
import { BiCartAlt } from 'react-icons/bi';

export const ItemCard = ({ cart }) => (
  <div className='itemCart l-flexCenterAround'>
    <span className='itemCart-icon'>
      <BiCartAlt />
    </span>
    <p className='itemCart-number' data-number={cart}></p>
  </div>
);
ItemCard.prototype = {
  cart: PropTypes.number,
};
