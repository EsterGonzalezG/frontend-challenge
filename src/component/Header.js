import React from 'react';

import { Title } from './Title';
import { Breadcrumbs } from './Breadcrumbs';
import {ItemCard}  from './ItemCart'
export const Header = () => (
    <div className='header l-marginBottom-24'>
        <div className='header-content'>
        <Title>Tienda de mÃ³biles</Title>
        <Breadcrumbs/>
        </div>
        <ItemCard/>
    </div>
)


