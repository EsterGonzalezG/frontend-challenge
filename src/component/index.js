import { setTitleDescription } from '../utils/setTitleDescription';
import { CartContext } from './../useContext/CartContext';
import { Breadcrumbs } from './Breadcrumbs';
import { ButtonBackToHome } from './ButtonBackToHome';
import { Image } from './Image';
import { ItemCard } from './ItemCart';
import { NotData } from './NotData';
import { Search } from './Search';
import { Select } from './Select';
import { Title } from './Title';

export {
  ButtonBackToHome,
  Select,
  Breadcrumbs,
  ItemCard,
  Image,
  Search,
  CartContext,
  NotData,
  Title,
  setTitleDescription,
};
