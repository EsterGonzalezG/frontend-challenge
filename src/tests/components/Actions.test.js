import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import { Actions } from '../../component/Actions';

describe('Test Actions component', () => {
  const setColor = jest.fn();
  const setStorage = jest.fn();
  const addProductCart = jest.fn();
  const product = {};

  const wrapper = shallow(
    <Actions product={product} setColor={setColor} setStorage={setStorage} addProductCart={addProductCart} />,
  );
  test('should be render a component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should not be handleSubmit', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
  });
  test('should be change input and called setValueSearch function', () => {
    const button = wrapper.find('button');
    button.simulate('click');
    expect(addProductCart).toHaveBeenCalled();
  });
});
