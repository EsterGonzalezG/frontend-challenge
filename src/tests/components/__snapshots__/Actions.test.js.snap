// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search component should be render a component 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    className="actions-label l-flexCenterAround l-marginBottom-24"
  >
    <Memo()
      name="Color"
      setOptionsValue={[MockFunction]}
    />
    <Memo()
      name="Storage"
      setOptionsValue={[MockFunction]}
    />
  </label>
  <div
    className="l-flexCenterAround"
  >
    <button
      className="Button Button--primary"
      onClick={[MockFunction]}
      type="button"
    >
      Add
    </button>
    <ButtonBackToHome>
      List
    </ButtonBackToHome>
  </div>
</form>
`;
